source/user/dft.rst

```diff
@@ -6,495 +6,145 @@ Density functional theory (DFT)
-
-* A functional name can have at most one factor.  If the factor is not given, it
-  is set to 1.  Compound functionals can be scaled as a unit. For example
-  '0.5*b3lyp' is equivalent to 'HF*0.1 + .04*LDA + .36*B88, .405*LYP + .095*VWN'
-
-* String "HF" stands for exact exchange (HF K matrix).  "HF" can be put in the
-  correlation functional part (after the comma). Putting "HF" in the correlation
-  part is the same as putting "HF" in the exchange part.
-
-* String "RSH" means range-separated operator. Its format is RSH(alpha; beta;
-  omega).  Another way to input RSH is to use keywords SR_HF and LR_HF:
-  "SR_HF(0.1) * alpha_plus_beta" and "LR_HF(0.1) * alpha" where the number in
-  parenthesis is the value of omega.
-
-* Be careful with the libxc convention of GGA functionals, in which the LDA
-  contribution is included.
-
-
-There is also another way to customize XC functionals which uses the :py:meth:`eval_xc`
-method of the numerical integral class::
-
-    mol = gto.M(atom='H 0 0 0; F 0.9 0 0', basis = '6-31g')
-    mf = dft.RKS(mol)
-    def eval_xc(xc_code, rho, spin=0, relativity=0, deriv=1, verbose=None):
-        # A fictitious XC functional to demonstrate the usage
-        rho0, dx, dy, dz = rho
-        gamma = (dx**2 + dy**2 + dz**2)
-        exc = .01 * rho0**2 + .02 * (gamma+.001)**.5
-        vrho = .01 * 2 * rho0
-        vgamma = .02 * .5 * (gamma+.001)**(-.5)
-        vlapl = None
-        vtau = None
-        vxc = (vrho, vgamma, vlapl, vtau)
-        fxc = None  # 2nd order functional derivative
-        kxc = None  # 3rd order functional derivative
-        return exc, vxc, fxc, kxc
-    dft.libxc.define_xc_(mf._numint, eval_xc, xctype='GGA')
-    mf.kernel()
-
-By calling the :func:`dft.libxc.define_xc_` function, the customized :func:`eval_xc`
-function is patched to the numerical integration class :attr:`mf._numint`
-dynamically.
-
-More examples of DFT XC functional customization can be found in
-:source:`examples/dft/24-custom_xc_functional.py` and
-:source:`examples/dft/24-define_xc_functional.py`.
+    E = T_s + E_{\rm ext} + E_J + E_{\rm xc} \ .
+
+Here, :math:`T_s` is the noninteracting kinetic energy, :math:`E_{\rm ext}` is the energy due to the external potential, :math:`E_J` is the Coulomb energy, and
+:math:`E_{\rm xc}` is the exchange-correlation (*xc*) energy. In practice, :math:`E_{\rm xc}` is approximated by a density functional approximation, which themselves may be divided into several classes along different rungs of Jacob's ladder :cite:`perdew_jacobs_ladder_aip_conf_proc_2001`:
+
+* local density approximations (e.g. LDA; *xc* energy depends only on the electron density, :math:`\rho`), 
+* generalized gradient approximations (GGA; *xc* energy also depends on the density gradient, :math:`|\nabla\rho|`), 
+* meta-GGAs (*xc* energy also depends on the kinetic energy density and Laplacian, :math:`\sum_i |\nabla \psi_i|^2`, :math:`\nabla^2\rho`),
```

Comment:
It should be noted that Laplacian density functionals are not supported in PySCF.

Reply:
Please see commit e451415

Reply:
OK